Upgrade to Halogen 0.5.0; fix #24

jonsterling opened the pull request:
- upgrade dependencies and build infrastructure
- support new component model
- fix warnings, minor cleanup

~~:zap: This is ready for review, but not merge; once `purescript-halogen v0.5.0` is released, I will update this PR accordingly. :zap:~~

garyb commented on docs/Data/Json/JTable.md:
````diff
@@ -3,19 +3,19 @@
 
 ``` purescript
-noStyle :: TableStyle
+noStyle :: forall p. TableStyle p
````
Is this due to my recent changes? :/

jonsterling replied:
Nah, it's unrelated :) I just needed this to be open-ended within the module it was defined.

jonsterling commented:
I've updated this PR to adapt to last night's Halogen changes, and also simplified the types that I had made more complicated before. Likewise, this will be ready to merge after this PR is updated to reflect the (impending) Halogen 0.5.0 release.

jonsterling commented:
OK, I think that this is ready for final review & merge, now that I have updated it to point at the released version of Halogen.

cryogenian commented on src/Data/Json/JTable/Internal.purs:
```diff
@@ -84,48 +93,54 @@ toPrim = foldJsonPrim Just (const Nothing) (const Nothing)
 enumerate xs = zipWith Tuple xs (range zero $ (length xs - one))
 
-renderJTableRaw :: JTableOpts -> Json -> Markup
+renderJTableRaw :: forall a. JTableOpts -> Json -> Markup
```
`forall a.`

jonsterling replied:
thank you for catching it!

cryogenian commented:
:+1:

jonsterling commented:
Just realized I need to update the README.
